python/runNanoSeq.py

```diff
@@ -0,0 +1,938 @@
+      iofile.write(str(njobs ))
+
+  #execute indeliantcaller commans
+  print("Starting indel calculation\n")
+  if ( args.index == None ) :
+    #multithread
+    with Pool( args.threads ) as p :
+      p.starmap( runCommand, commands )
+  else :
+    #array execution
+    runCommand( commands[args.index - 1][0] )
+  print("Completed indel calculation\n")
+
+#indel section
+if (args.subcommand == 'post' ) :
+  if ( os.path.isfile(tmpDir+'/post/1.done') ) : sys.exit(0)
+  if ( args.index != None and args.index > 1 ) :
+    print("\nWarning can only use 1 job of array\n")
+    sys.exit(0)
+
+  #check dsa
+  if (not os.path.isfile(tmpDir+'/dsa/nfiles') ):
+    sys.exit(tmpDir+'/dsa/nfiles not found!\n')
+  else :
+    with open(tmpDir+'/dsa/nfiles') as iofile :
+      nfiles = int(iofile.readline())
+
+  for i in range(nfiles ) :
+    if ( len(glob.glob(tmpDir+"/dsa/%s.done"%i)) != 1 ) :
+      sys.exit("\ndsa job %s did not complete correctly\n"%i)
+    if ( len(glob.glob(tmpDir+"/dsa/%s.dsa.bed.gz"%i)) != 1 ) :
+      sys.exit("\ndsa job %s did not complete correctly\n"%i)
+
+  #check var
+  did_var = False
+  if ( os.path.isfile(tmpDir+'/var/nfiles') ):
+    did_var = True
+    with open(tmpDir+'/dsa/nfiles') as iofile :
+      nfiles = int(iofile.readline())
+    for i in range(nfiles ) :
+      if ( len(glob.glob(tmpDir+"/var/%s.done"%i)) != 1 ) :
+        sys.exit("\nvar job %s did not complete correctly\n"%i)
+      if ( len(glob.glob(tmpDir+"/var/%s.var"%i)) != 1 ) :
+        sys.exit("\nvar job %s did not complete correctly\n"%i)
+      if ( len(glob.glob(tmpDir+"/var/%s.cov.bed.gz"%i)) != 1 ) :
+        sys.exit("\nvar job %s did not complete correctly\n"%i)
+
+  #check indel
+  did_indel = False
+  if ( os.path.isfile(tmpDir+'/indel/nfiles') ):
+    did_indel = True
+    with open(tmpDir+'/indel/nfiles') as iofile :
+      nfiles = int(iofile.readline())
+    for i in range(nfiles ) :
+      if ( len(glob.glob(tmpDir+"/indel/%s.done"%i)) != 1 ) :
+        sys.exit("\nindel job %s did not complete correctly\n"%i)
+      if ( len(glob.glob(tmpDir+"/indel/%s.indel.filtered.vcf.gz"%i)) != 1 ) :
+        sys.exit("\nindel job %s did not complete correctly\n"%i)
+
+
+  if ( did_var ) :
+    #generate csv files
+    print("\nGenerating CSV files for var\n")
+    csvFiles = {
+                'Coverage'   : open('%s/post/%s' % (tmpDir, 'coverage.csv'), 'w'),
+                'CallVsQpos' : open('%s/post/%s' % (tmpDir, 'callvsqpos.csv'), 'w'),
+                'PyrVsMask'  : open('%s/post/%s' % (tmpDir, 'pyrvsmask.csv'), 'w'),
+                'ReadBundles': open('%s/post/%s' % (tmpDir, 'readbundles.csv'), 'w'),
+                'Burdens'    : open('%s/post/%s' % (tmpDir, 'burdens.csv'), 'w'),
+                'Variants'   : open('%s/post/%s' % (tmpDir, 'variants.csv'), 'w'),
+                'Mismatches' : open('%s/post/%s' % (tmpDir, 'mismatches.csv'), 'w')
+               }
+    #write headers
+    csvFiles['Coverage'].write('count\n')
```

Comment:
Can csv file generation be abstracted to a function.  Aim to have a high level function that just calls others to make it easier to read and maintain individual elements of funtionality.